### DISCUSSION TITLE
feat: Ability to trigger campaigns via API

### DISCUSSION THREAD

**Discussion by sojan-official** [chatwoot/chatwoot#8467] [Feature Requests]
ref: https://discord.com/channels/647412545203994635/647729576427913228/943207330487615498

> is there any way I could start a new campaign via an API interaction
> I was planning to integrate with the instatus API to start a campaign that states the current status of the site

------ 

The ability to trigger campaigns via API will open up more room for unique use cases. 

## Current limitations 

-  Campaigns will be triggered only if no conversation exists for the contact 
-  Campaigns trigger logic happens in frontend based on page rules in case of web widget channel 

## Thoughts
-  The messages like site status or a product update note doesn't need to persist unless the user triggers a conversation over it. 

ref: #3997 https://github.com/chatwoot/chatwoot/issues/2768

**Comment by sojan-official** [comment]
- the specific use case of sharing a downtime alert could be solved using one-off SMS campaigns which can be created via APIs

**Comment by brundles** [comment]
Looking forward to this. Do you have any ETA?

**Comment by pranavrajs** [comment]
I'm moving these to discussions as we will not be prioritizing these immediately.

**Comment by luiszxcv** [comment]
News? !Important.

**Comment by basebright** [comment]
any news about it? sounds like an enabler for a lot of useful scenarios
